lib/credential-pair.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * A Credential Pair combines a secret element and a public element. The public
+ * element is stored in an SSM Parameter, while the secret element is stored in
+ * AWS Secrets Manager.
+ *
+ * For example, this can be:
+ * - A username and a password
+ * - A private key and a certificate
+ * - An OpenPGP Private key and its public part
+ */
+export interface ICredentialPair {
+  /**
+   * The ARN of the SSM parameter containing the public part of this credential
+   * pair.
+   */
+  readonly parameterArn: string;
```

Comment:
Again, the word "public" and "private" should appear in these property names. Ideally the type should not be just strings but rather `ssm.IParameter` and `secretsmanager.ISecret`, but I don't think we have those, right?

Reply:
Yeah I don't reckon we have those either.